### PR TITLE
CLN: address TODOs

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/formats/style/test_html.py [hunk 1]
```diff
@@ -488,8 +489,12 @@ def test_replaced_css_class_names(styler_mi):
...
     ).set_table_styles(css_class_names=css)
     styler_mi.index.names = ["n1", "n2"]
-    styler_mi.hide_index(styler_mi.index[1:])
-    styler_mi.hide_columns(styler_mi.columns[1:])
+
+    msg = "this method is deprecated in favour of `Styler.hide"
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        styler_mi.hide_index(styler_mi.index[1:])
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        styler_mi.hide_columns(styler_mi.columns[1:])
```
Comment (attack68) [inline pandas/tests/io/formats/style/test_html.py:497]:
```suggestion
styler_mi.hide(styler_mi.index[1:], axis=0)
styler_mi.hide(styler_mi.columns[1:], axis=1)
```

would it be better to replace these tests with the non-deprecated version?

File: pandas/tests/io/formats/style/test_html.py [hunk 2]
```diff
@@ -611,9 +616,13 @@ def test_hiding_index_columns_multiindex_alignment():
...
-    styler.hide_index(level=1).hide_columns(level=0)
-    styler.hide_index([("j0", "i1", "j2")])
-    styler.hide_columns([("c0", "d1", "d2")])
+    msg = "this method is deprecated in favour of `Styler.hide"
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        styler.hide_index(level=1).hide_columns(level=0)
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        styler.hide_index([("j0", "i1", "j2")])
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        styler.hide_columns([("c0", "d1", "d2")])
```
Comment (attack68) [inline pandas/tests/io/formats/style/test_html.py:625]:
```suggestion
styler.hide(level=1, axis=0).hide(level=0, axis=1)
styler.hide([("j0", "i1", "j2")], axis=0)
styler.hide([("c0", "d1", "d2")], axis=1)
```

File: pandas/tests/io/formats/style/test_style.py [hunk 3]
```diff
@@ -1530,7 +1530,9 @@ def test_hiding_headers_over_index_no_sparsify():
     df = DataFrame(9, index=midx, columns=[0])
     ctx = df.style._translate(False, False)
     assert len(ctx["body"]) == 6
-    ctx = df.style.hide_index((1, "a"))._translate(False, False)
+    msg = "this method is deprecated in favour of `Styler.hide"
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        ctx = df.style.hide_index((1, "a"))._translate(False, False)
```
Comment (attack68) [inline pandas/tests/io/formats/style/test_style.py:1535]:
```suggestion
    ctx = df.style.hide((1, "a"), axis=0)._translate(False, False)
```

File: pandas/tests/io/formats/style/test_to_latex.py [hunk 4]
```diff
@@ -803,7 +804,9 @@ def test_css_convert_apply_index(styler, axis):
 
 def test_hide_index_latex(styler):
     # GH 43637
-    styler.hide_index([0])
+    msg = "this method is deprecated in favour of `Styler.hide"
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        styler.hide_index([0])
```
Comment (attack68) [inline pandas/tests/io/formats/style/test_to_latex.py:809]:
```suggestion
    styler.hide([0], axis=0)
```

File: pandas/tests/io/formats/style/test_to_latex.py [hunk 5]
```diff
@@ -826,9 +829,14 @@ def test_latex_hiding_index_columns_multiindex_alignment():
...
-    styler.hide_index([("i0", "i1", "i2")])
-    styler.hide_columns([("c0", "c1", "c2")])
+
+    msg = "this method is deprecated in favour of `Styler.hide"
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        styler.hide_index(level=1).hide_columns(level=0)
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        styler.hide_index([("i0", "i1", "i2")])
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        styler.hide_columns([("c0", "c1", "c2")])
```
Comment (attack68) [inline pandas/tests/io/formats/style/test_to_latex.py:839]:
```suggestion
        styler.hide(level=1, axis=0).hide(level=0, axis=1)
        styler.hide([("i0", "i1", "i2")], axis=0)
        styler.hide([("c0", "c1", "c2")], axis=1)
```

**Review by jreback — APPROVED** [review]
lgtm merge on green

**Pull request merged**